src/middlewared/middlewared/plugins/asigra.py

```diff
@@ -0,0 +1,358 @@
+import glob
+import os
+import psycopg2 as pg
+import re
+import shutil
+import subprocess
+
+from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT
+
+from middlewared.service import Service
+from middlewared.utils import Popen
+
+
+class AsigraService(Service):
+
+    class Config:
+        service = "asigra"
+        datastore_prefix = "asigra_"
+
+    def __init__(self, middleware):
+        super(AsigraService, self).__init__(middleware)
+
+        self.pg_home = "/usr/local/pgsql"
+        self.pg_user = "pgsql"
+        self.pg_group = "pgsql"
+
+        self.dssystem_path = "/usr/local/ds-system"
+        self.dssystem_db_path = os.path.join(self.dssystem_path, "db")
+
+    # Use etc plugin for this?
+    async def setup_postgresql(self):
```

Comment:
Given all the file and blocking operations this should not be a coroutine, but a normal python method.